packedForest/src/forestTypes/binnedTree/binStruct.h

```diff
@@ -383,7 +384,44 @@ namespace fp{
 				}
 
 
-				//inline int predictObservation(std::vector<T>& observation, identity<int>){
+				inline void predictBinObservation(int observationNum, std::vector<int>& preds, identity<weightedFeature>){
```

Comment:
do you have any ideas for how we could reduce the amount of repeated code for each of the different predict function forest types

Reply:
Sure, in the future we can do away with everything but rerf since RF can be run by choosing the right sparse matrix for RerF.  This is future work though.